libs/server/Resp/Parser/SessionParseState.cs

```diff
@@ -167,11 +168,11 @@ public void InitializeWithArguments(ArgSlice arg1, ArgSlice arg2, ArgSlice arg3,
         /// </summary>
         /// <param name="args">Set of arguments to initialize buffer with</param>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public void InitializeWithArguments(ArgSlice[] args)
+        public void InitializeWithArguments(IList<ArgSlice> args)
```

Comment:
This change incurs overhead when this is called with an array. Please just call ToArray() on your list.

Reply:
@badrishc  Tal and I spoke about this and would like your opinion on this.

My understanding is that this will be same cost as a lookup call on  a concrete class that we were passing before.

We dont have to go through the vtable lookup chain because of this section of the CLR book:

https://sakupinera.github.io/2022/11/04/csharp/clr-via-csharp/Chapter%2013%20Interfaces/

I may be wrong but if the type is there in my method table on the first look, is that not same cost as calling a method on a concrete class for the first time?

Reply:
Would the below version then be better?
`       [MethodImpl(MethodImplOptions.AggressiveInlining)]
        public void InitializeWithArguments<T>(T args) where T : IList<ArgSlice>
        {
            Initialize(args.Count);

            for (var i = 0; i < args.Count; i++)
            {
                *(bufferPtr + i) = args[i];
            }
        }`

Reply:
Tal is confirmed, I decompiled to IL and I can confirm it always uses callvirt with instance